samcli/lib/telemetry/metric.py

```diff
@@ -148,16 +148,18 @@ def wrapped(*args, **kwargs):
             # Execute the function and capture return value. This is returned back by the wrapper
             # First argument of all commands should be the Context
             return_value = func(*args, **kwargs)
-        except UserException as ex:
+        except (UserException, click.BadOptionUsage, click.BadArgumentUsage, click.BadParameter) as ex:
             # Capture exception information and re-raise it later so we can first send metrics.
             exception = ex
-            exit_code = ex.exit_code
-            if ex.wrapped_from is None:
-                exit_reason = type(ex).__name__
-            else:
+            # TODO (hawflau): review exitcode meaning. We now understand exitcode 1 as errors fixable by users.
+            exit_code = 1
+            if hasattr(ex, "wrapped_from") and ex.wrapped_from:
                 exit_reason = ex.wrapped_from
+            else:
+                exit_reason = type(ex).__name__
         except Exception as ex:
-            exception = ex
+            command = ctx.command_path if ctx else ""
```

Comment:
Is there any normal case when `ctx` is None? Or is this just for weird edge cases?

Reply:
> Or is this just for weird edge cases?
yep. it shouldn't happen at all